src/overlays/actors/ovl_En_Takaraya/z_en_takaraya.c

```diff
@@ -28,61 +47,360 @@ ActorInit En_Takaraya_InitVars = {
+            D_80ADFB10[i] = Lib_SegmentedToVirtual(D_80ADFB10[i]);
+        }
+        D_80ADFB28 = 1;
+    }
+    this->eyeTexIndex = 2;
+    if (gSaveContext.save.entrance == ENTRANCE(TREASURE_CHEST_SHOP, 1)) {
+        Audio_StopSubBgm();
+        if (gSaveContext.timerStates[TIMER_ID_MINIGAME_2] == TIMER_STATE_6) {
+            this->actor.textId = 0x77A;
+            gSaveContext.timerStates[TIMER_ID_MINIGAME_2] = TIMER_STATE_STOP;
+            func_80ADF6DC(this);
+        } else {
+            this->actor.textId = 0x77C;
+            if (CHECK_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED)) {
+                CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_WAIT);
+                CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED);
+                func_80ADEDF8(this);
+            } else {
+                func_80ADF6DC(this);
+            }
+        }
+    } else {
+        func_80ADEDF8(this);
+    }
+}
+
+void EnTakaraya_Destroy(Actor* thisx, PlayState* play) {
+    EnTakaraya* this = THIS;
+
+    Flags_UnsetSwitch(play, 5);
+    if (!this->unk2AD) {
+        CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_WAIT);
+        gSaveContext.timerStates[TIMER_ID_MINIGAME_2] = TIMER_STATE_STOP;
+    }
+}
+
+void func_80ADED8C(EnTakaraya* this) {
+    if (this->eyeTexIndex != 0) {
+        this->eyeTexIndex++;
+        if (this->eyeTexIndex == 4) {
+            this->eyeTexIndex = 0;
+        }
+    } else if (Rand_ZeroOne() < 0.02f) {
+        this->eyeTexIndex++;
+    }
+}
+
+void func_80ADEDF8(EnTakaraya* this) {
+    if (this->skelAnime.animation == &object_bg_Anim_001384) {
+        Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_00AD98, 5.0f);
+    }
+    this->actionFunc = func_80ADEE4C;
+}
+
+void func_80ADEE4C(EnTakaraya* this, PlayState* play) {
+    u16 textId;
+
+    if (SkelAnime_Update(&this->skelAnime)) {
+        if ((&object_bg_Anim_00A280) == (*this).skelAnime.animation) {
+            Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_00AD98, 5.0f);
+        } else {
+            Animation_MorphToLoop(&this->skelAnime, &object_bg_Anim_009890, -4.0f);
+        }
+    }
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        if (Text_GetFaceReaction(play, FACE_REACTION_SET_TREASURE_CHEST_SHOP_GAL) == 0) {
+            Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_00A280, -4.0f);
+        }
+        func_80ADF03C(this);
+    } else if (Actor_IsFacingPlayer(&this->actor, 0x2000)) {
+        textId = Text_GetFaceReaction(play, FACE_REACTION_SET_TREASURE_CHEST_SHOP_GAL);
+        this->actor.textId = textId;
+        if (!(textId & 0xFFFF)) {
+            this->actor.textId = D_80ADFB2C[(void)0, gSaveContext.save.playerForm];
+        }
+        this->unk2B0 = gSaveContext.save.playerForm + this->switchFlag;
+        Actor_OfferTalk(&this->actor, play, 120.0f);
+    }
+}
+
+void func_80ADEF74(EnTakaraya* this, PlayState* play) {
+    u8 var_v1;
+
+    if (Flags_GetSwitch(play, this->unk2B0)) {
```

Comment:
```suggestion
    if (Flags_GetSwitch(play, this->formSwitchFlag)) {
```
Above this
```
this->unk2B0 = gSaveContext.save.playerForm + this->switchFlag;
```
We see that which switch flag we use for determining if the player has finished the game is determined by the playerform.